src/main/java/org/apache/mesos/scheduler/plan/DefaultStageScheduler.java

```diff
@@ -14,47 +14,63 @@
 import java.util.List;
 
 /**
- * Default scheduler.
- * Attempts to meet the Offer requirements of the block presented, and perform the appropriate Operations
+ * Default scheduler. See docs in {@link StageScheduler} interface.
  */
 public class DefaultStageScheduler implements StageScheduler {
 
-  private final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final Logger logger = LoggerFactory
+            .getLogger(DefaultStageScheduler.class);
+    private final OfferAccepter offerAccepter;
+    private final OfferEvaluator offerEvaluator;
 
-  private OfferAccepter offerAccepter = null;
+    @Inject
+    public DefaultStageScheduler(OfferAccepter offerAccepter) {
+        this(offerAccepter, new OfferEvaluator());
+    }
 
-  @Inject
-  public DefaultStageScheduler(OfferAccepter offerAccepter) {
-    this.offerAccepter = offerAccepter;
-  }
+    public DefaultStageScheduler(OfferAccepter offerAccepter, OfferEvaluator offerEvaluator) {
```

Comment:
Since an OfferEvaluator is constructed with an OfferRequirement, this implies that we need to reconstruct this Scheduler on every Offer cycle.  I'm not sure that's a good idea.  Why was this done?